### ISSUE TITLE
Admin plugin dirties environment overrides

### ISSUE THREAD

**Issue by ginkgomzd** [getgrav/grav-plugin-admin#1614]
It was suggested in #27 to make a selection of which config should be updated when changing configurations in the admin. It appears to only have gone as far as displaying which config will be edited.

It seems like a pretty legitimate use-case to edit production/default environment configs in a development environment.

There is another problem. The grav docs on environment config ( https://learn.getgrav.org/advanced/environment-config) state, "Then you can override this with a setting that _**only overrides specific options**_ that you want to use for local testing:".
This works fine, but if you go and use the Admin plugin to change configurations, then all configuration options are written to the environment-specific config..

In summary, either:
* Prevent admin from adding configs to environment-specific configs. The UX could be to disable editing of any configs that are not declared in the active configuration. Some logic would be needed determine which configs are okay to write to file.
* Allow editing of only the default config, i.e. do not allow editing any environment-specific configs with the Admin.
* Provide the ability to select which environment config should be updated in the Admin UI.

**Comment by rhukster** [comment]
This is a known (and long-standing issue).  The environment overrides are quite an advanced feature and works great when working directly with the files. Unfortunately creating a UI in the admin that can control which options are stored where, is a very complicated undertaking. And changing the default behavior is not really an option at this point.

We really are at a point where we don't want to radically redesign the current admin (which is what would be required to solve this). However we have plans to redevelop the admin from scratch for Grav 2.0, and at that time, we'll take a look at this issue again, and see if we can come up with a solution, and take your ideas into account.

**Comment by ginkgomzd** [comment]
Thanks for the reply.

Is there any place for discussions about the 2.0 roadmap and architecture?

**Comment by rhukster** [comment]
Yup, i've now 'pinned it' : https://github.com/getgrav/grav/issues/1767

**Comment by lufog** [comment]
I would like to see сhecking for updates to themes/plugins by name and author. Since not all extensions written for grav are published in the repository. I had a case of a client site crash after updating Grav and plugins. It turned out that the plug-in written by me, for his website, when updated, was replaced by another from the repository. At that time, lucky, because i had configured backup on a schedule.
